### PR TITLE
Adding rbd python tests to cephci

### PR DESCRIPTION
**Pull request by manasagowri**
Automation of Test Case CEPH-83574791 - Create random different data and perform writes using rbd python module which provides file access to rbd images and verify

Test Steps
1. create RBD pool and image
2. create random different data and perform writes using rbd python module which provides file access to rbd images and verify

Success Logs - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-67LCPF/

### CODE REVIEW

File: suites/reef/rbd/tier-2_rbd_regression.yaml [hunk 1]
```diff
@@ -199,3 +199,17 @@ tests:
       module: test_rbd_compression.py
       name: Test to verify data compression on global, pool and image level
       polarion-id: CEPH-83574644
+
+  - test:
```
Comment (sunilangadi2) [inline suites/reef/rbd/tier-2_rbd_regression.yaml:203]: This test would be applicable to pacific and quincy, as well Is it?

File: tests/rbd/test_rbd_python_module.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    which provides file access to rbd images and verify
+
+    """
+    log.info(
+        "Running python rbd module image creation, write and read data - CEPH-83574791"
+    )
+
+    try:
+        client = kw.get("ceph_cluster").get_ceph_object(role="client").node
+        client.exec_command(cmd="sudo pip install docopt")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_python_module.py:63]: `docopt` is already there here https://github.com/red-hat-storage/cephci/blob/5c76b84270023f8a9b159921108190b618bc94c8/setup.py#L19
why to install again, also other test are already using it.

File: tests/rbd/test_rbd_python_module.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                    cmd += f" --data-pool {data_pool}"
+                out, err = client.exec_command(cmd=cmd)
+                if err or "don't match" in out:
+                    log.error(out)
+                    log.error(
+                        f"Executing rbd python to create image, read and\
+                               write data failed for image {pool}/{image}"
+                    )
+                    return 1
+                log.info(out)
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_python_module.py:35]: please add `log.info` for success message as well.

File: ceph/rbd/workflows/rbd_python.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def rbd_python(args):
+    """
+    Create image and write random data to it, read data written and verify
+    """
+    conf_file = args.get("--conf-file", "/etc/ceph/ceph.conf")
+    pool = args.get("--pool")
+    image_name = args.get("--image")
+    image_size = int(args.get("--image-size"))
```
Comment (sunilangadi2) [inline ceph/rbd/workflows/rbd_python.py:34]: Image size should add with data byte parameters as well right like `Mb` `G` `T`

File: ceph/rbd/workflows/rbd_python.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    data_pool = args.get("--data-pool")
+
+    with rados.Rados(conffile=conf_file) as cluster:
```
Comment (sunilangadi2) [inline ceph/rbd/workflows/rbd_python.py:37]: please add comment for what we are doing here?

File: ceph/rbd/workflows/rbd_python.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        with cluster.open_ioctx(pool) as ioctx:
+            rbd_inst = rbd.RBD()
+            if data_pool:
+                rbd_inst.create(ioctx, image_name, image_size, data_pool=data_pool)
+            else:
+                rbd_inst.create(ioctx, image_name, image_size)
+            with rbd.Image(ioctx, image_name) as image:
+                data = str.encode("bar" * 20000)
+                image.write(data, 0)
```
Comment (sunilangadi2) [inline ceph/rbd/workflows/rbd_python.py:46]: how much size data gets writes here?

File: ceph/rbd/workflows/rbd_python.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                mydata = image.read(0, 60000)
+                if data != mydata:
```
Comment (sunilangadi2) [inline ceph/rbd/workflows/rbd_python.py:48]: not cleared about data validation why `60000` for read? Does it bar 3*20000?

File: ceph/rbd/workflows/rbd_python.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                    print(
+                        f"Data written and read back don't match for image {image_name}"
+                    )
+                    return 1
+    print(f"Data written matches with the data read back for image {image_name}")
+    return 0
+
+
+if __name__ == "__main__":
+    args = docopt(doc)
```
Comment (sunilangadi2) [inline ceph/rbd/workflows/rbd_python.py:58]: make `help=True` in beginning itself so it's easy for users.
Ref: https://github.com/red-hat-storage/cephci/blob/5c76b84270023f8a9b159921108190b618bc94c8/pipeline/scripts/ci/upstream_cli.py#L206

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: tests/rbd/test_rbd_python_module.py (hunk 2 above)
Comment (manasagowri) [reply tests/rbd/test_rbd_python_module.py:63]: docopt will be installed in jenkins execution node as part of this. But I am executing rbd python tests inside the client node where again I need docopt, that's why this command will install docopt inside our cluster's client node.

File: tests/rbd/test_rbd_python_module.py (hunk 3 above)
Comment (manasagowri) [reply tests/rbd/test_rbd_python_module.py:35]: this log.info out will actually print the success message itself clearly if you check the logs for this execution.

File: ceph/rbd/workflows/rbd_python.py (hunk 4 above)
Comment (manasagowri) [reply ceph/rbd/workflows/rbd_python.py:34]: No in rbd python we need to specify size in bytes as an integer itself. This is different from the way we pass in CLI command to create image.

File: ceph/rbd/workflows/rbd_python.py (hunk 6 above)
Comment (manasagowri) [reply ceph/rbd/workflows/rbd_python.py:46]: around 4MB data is written

File: ceph/rbd/workflows/rbd_python.py (hunk 7 above)
Comment (manasagowri) [reply ceph/rbd/workflows/rbd_python.py:48]: bar is 3 letters and that multipied by 20000 will actually be 60000 letters written that's why we are reading 60000 back to compare

File: suites/reef/rbd/tier-2_rbd_regression.yaml (hunk 1 above)
Comment (manasagowri) [reply suites/reef/rbd/tier-2_rbd_regression.yaml:203]: Added test to pacific and quincy suite as well.

File: ceph/rbd/workflows/rbd_python.py (hunk 5 above)
Comment (manasagowri) [reply ceph/rbd/workflows/rbd_python.py:38]: Done.

File: ceph/rbd/workflows/rbd_python.py (hunk 8 above)
Comment (manasagowri) [reply ceph/rbd/workflows/rbd_python.py]: Done.

**Review by sunilangadi2 — APPROVED** [review]
changes LGTM.

**Pull request merged**